### ISSUE TITLE
Consider upgrading to Jinja 3.X

### ISSUE THREAD

**Issue by webb-ben** [geopython/pygeoapi#715]
**Description**

As described in pallets/jinja#827, the urlize() function incorrectly removes a trailing close parentheses from URLs when creating a hyperlink in the HTML landing pages.

**Steps to Reproduce**

- To view in pygeoapi with Docker: 
1) Run `docker run -p 5000:80 --rm webbben/pygeoapi-sensorthings:1.2.2` which is an example STA pygeoapi endpoint with many links ending in close parentheses.
2) Navigate to [http://localhost:5000/collections/Datastreams/items/8862231](http://localhost:5000/collections/Datastreams/items/8862231)
![image](https://user-images.githubusercontent.com/40066515/122231953-df2bdd00-ce88-11eb-89f6-d6dee7834b50.png)
As can be seen in the bottom left, and the `@iot.selfLink`, the trailing `)` is dropped from the hyperlink.
- To view in python, running in the virtual environment that pygeoapi is running on:
```
from jinja2.utils import urlize
example_url = u'https://wikipedia.org/wiki/geopython(pygeoapi)'
urlize(example_url)
```
With (default, Jinja2==2.10.1): `urlize(example_url)` -> `'<a ref="https://wikipedia.org/wiki/geopython(pygeoapi"> https://wikipedia.org/wiki/geopython(pygeoapi</a>)'`

**Expected behavior**
![image](https://user-images.githubusercontent.com/40066515/122237351-4350a000-ce8d-11eb-82ef-2b80b77c768c.png)

With (Jinja2==3.0.1): `urlize(example_url)` -> `'<a ref="https://wikipedia.org/wiki/geopython(pygeoapi)"> https://wikipedia.org/wiki/geopython(pygeoapi)</a>'`

**Environment**

 - OS: Any
 - Python version: 3
 - pygeoapi version: master/latest

**Additional context**

This is a problem with Jinja2 that has already been fixed in v3.X which was released in May of 2021. I am not aware of outside reasons to wait before upgrading to the newest version of Jinja2, as v2.10.1 installed as the default dependency with Flask.

**Comment by tomkralidis** [comment]
cc @kalxas 

Thanks for the report.  I think we need to wait until Jinja2 3.x makes its way to Ubuntu/Debian packaging.

**Comment by tomkralidis** [comment]
@webb-ben is this now fixed, given #876?
